src/main/java/de/fraunhofer/aisec/cpg/graph/BinaryOperator.java

```diff
@@ -51,6 +50,11 @@
   /** The operator code. */
   private String operatorCode;
 
+  /** Required for compound BinaryOperators. This should not be stored in the graph */
+  @Transient
+  private final List<String> compoundOperators =
+      new ArrayList<>(Arrays.asList("*=", "/=", "%=", "+=", "-=", "<<=", ">>=", "&=", "^=", "|="));
```

Comment:
This creates a partly immutable list and then a (mutable) ArrayList from it. Why not directly instantiate the type that is needed?